alpha/lib/model/accessControl.php

```diff
@@ -156,7 +156,20 @@ public function hasRules($contextType = null)
 		}
 		return false;
 	}
-	
+
+	public function hasKESRule()
+	{
+		$rules = $this->getRulesArray();
+
+		foreach($rules as $rule)
+		{
+			/* @var $rule kRule */
+			if (json_decode($rule->getRuleData())->type == 'ServeFromKESRule')
```

Comment:
i think it's better to check if one of the action is KalturaAccessControlServeRemoteEdgeServerAction

+ return false at the end of the method

Reply:
also for performance issue, let's save the result in the custom_data when you save the ACP, and just check the custom_data in the action